app/lib/three_scale/api/collection.rb

```diff
@@ -1,42 +1,63 @@
-class ThreeScale::Api::Collection
-
-  include PaginationHelper::ClassMethods
+# frozen_string_literal: true
 
+class ThreeScale::Api::Collection
   def initialize(collection, options = {})
     @collection = collection
     @options = options
   end
 
-  def to_xml(options = {})
-    builder = options[:builder] || ThreeScale::XML::Builder.new
-
-    root = options.delete(:root) || @options.delete(:root)
-    # merge options passed when creating collection with the ones passed now together with builder
-    options = options.merge(@options).merge(builder: builder, skip_instruct: true)
+  def to_xml(xml_options = {})
+    builder = xml_options[:builder] || ThreeScale::XML::Builder.new
+    root = xml_options.delete(:root) || options.delete(:root)
+    all_options = xml_options.merge(options).merge(builder: builder, skip_instruct: true)
 
-    builder.tag!(root, pagination) do |xml|
+    pagination_metadata = paginated? ? pagination_attrs : {}
+    builder.tag!(root, pagination_metadata) do |xml|
       each do |item|
-        item.to_xml(options)
+        item.to_xml(all_options)
       end
     end
 
     builder.to_xml
   end
 
+  def to_json(json_options = {})
+    pagination_metadata = paginated? ? pagination_attrs(wrapper: :metadata) : {}
+    MultiJson.dump collection.to_hash(json_options).merge(pagination_metadata)
+  end
+
   def method_missing(method_name, *arguments, &block)
-    collection = @collection.try(:represented) || @collection
-    collection.send(method_name, *arguments, &block)
+    represented.send(method_name, *arguments, &block)
   end
 
   # http://robots.thoughtbot.com/post/28335346416/always-define-respond-to-missing-when-overriding
 
   def respond_to_missing?(method_name, include_private = false)
-    @collection.respond_to?(method_name, include_private)
+    collection.respond_to?(method_name, include_private)
   end
```

Comment:
I am scared to touch this 💣 
![image](https://user-images.githubusercontent.com/11318903/100596521-8c2f7580-32fc-11eb-8d44-988c09c9dad1.png)